src/Cli/dotnet/Commands/Run/RunCommand.cs

```diff
@@ -587,7 +587,10 @@ public static RunCommand FromParseResult(ParseResult parseResult)
             }
 
             // If '-' is specified as the input file, read all text from stdin into a temporary file and use that as the entry point.
-            entryPointFilePath = Path.GetTempFileName();
+            // We create a new directory for each file so other files are not included in the compilation.
```

Comment:
The CI failed, revealing a bug here, so I'm fixing it in the last commit.

Previously, the implementation for running code from stdin put the code into a temp file but with this PR other files in the temp directory could be potentially included in the compilation. I've fixed that by putting the stdin code files into isolated temp directories instead. I've reused the same machinery we have for creating the artifacts folders. I think both these kinds of temp dirs (for artifacts and for stdin code) can reuse other features when implemented (periodical cleanup and customizing the temp path).

Reply:
Is there a risk that `Path.GetRandomFileName()` will collide with another one from a previous run? Are we able to proceed and do something reasonable in that case?

I am curious if we could just think of each user having their own "stdin app" and reusing the artifacts path for it across runs. However, I don't think it's urgent/necessary to do so.